qa/rpc-tests/quality_nodes.py

```diff
@@ -0,0 +1,357 @@
+            mark_logs("Certificate is {}".format(cert_1_epoch_0), self.nodes, DEBUG_MODE)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send certificate failed with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+            assert(False)
+
+        mark_logs("Check cert is in mempools", self.nodes, DEBUG_MODE)
+        assert_equal(True, cert_1_epoch_0 in self.nodes[0].getrawmempool())
+        cert1_mined = self.nodes[0].generate(2)[0]
+        assert_true(cert_1_epoch_0 in self.nodes[0].getblock(cert1_mined, True)['cert'])
+
+        # Create Cert2 with quality 100 and place it in node1
+        mark_logs("Create Cert2 with quality 100 and place it in node1", self.nodes, DEBUG_MODE)
+        pkh_node1 = self.nodes[1].getnewaddress("", True)
+        amount_cert_1 = [{"pubkeyhash": pkh_node1, "amount": bwt_amount}]
+        quality = 100
+        proof = mcTest.create_test_proof(
+            "sc1", epoch_number, epoch_block_hash, prev_epoch_block_hash,
+            quality, constant, [pkh_node1], [bwt_amount])
+        try:
+            cert_2_epoch_0 = self.nodes[1].send_certificate(scid, epoch_number, quality, epoch_block_hash, proof, amount_cert_1, CERT_FEE)
+            assert(len(cert_2_epoch_0) > 0)
+            mark_logs("Certificate is {}".format(cert_2_epoch_0), self.nodes, DEBUG_MODE)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send certificate failed with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+            assert(False)
+
+        #Cert should be in mempool and could be placed in node_1 block
+        mark_logs("Check cert is in mempools", self.nodes, DEBUG_MODE)
+        assert_equal(True, cert_2_epoch_0 in self.nodes[1].getrawmempool())
+        cert2_mined = self.nodes[1].generate(1)[0]
+        assert_true(cert_2_epoch_0 in self.nodes[1].getblock(cert2_mined, True)['cert'])
+
+        self.join_network()
+
+        self.sync_all()
+        assert_true(cert_1_epoch_0 in self.nodes[0].getblock(cert1_mined, True)['cert'])
+        #assert_false(cert_2_epoch_0 in self.nodes[0].getblock(cert2_mined, True)['cert'])
+        assert_false(cert_2_epoch_0 in self.nodes[0].getrawmempool())
+
+        self.nodes[0].generate(EPOCH_LENGTH - 2)
+        self.sync_all()
+
+        self.split_network()
+
+        # Create Cert3 with quality 100 and place it in node0
+        prev_epoch_block_hash = epoch_block_hash
+        epoch_block_hash, epoch_number = get_epoch_data(scid, self.nodes[0], EPOCH_LENGTH)
+        mark_logs("Height: {}. creation height {}".format(self.nodes[0].getblockcount(), sc_creating_height), self.nodes, DEBUG_MODE)
+        mark_logs("Height: {}. Epoch {}".format(self.nodes[0].getblockcount(), epoch_number), self.nodes, DEBUG_MODE)
+        mark_logs("Create Cert1 with quality 100 and place it in node0", self.nodes, DEBUG_MODE)
+        quality = 100
```

Comment:
I do not understand why this work. We have already confirmed at least one certificate with quality = 100, we should not be able to push another certificate with same quality neither in mempool nor in blockchain

Reply:
Quality value resets on each epoch. This certificate doesn't relate to the previous ones.